src/iosource/PktSrc.cc

```diff
@@ -109,15 +109,25 @@ void PktSrc::Opened(const Properties& arg_props)
 		return;
 		}
 
+
 	if ( props.is_live )
+		{
 		Info(fmt("listening on %s\n", props.path.c_str()));
 
+		// We only register the file descriptor if we're in live
+		// mode because libpcap's file descriptor for trace files
+		// isn't a reliable way to know whether we actually have
+		// data to read.
+		iosource_mgr->RegisterFd(props.selectable_fd, this);
```

Comment:
Can have a PktSrc be live, but without selectable FD, so need a check to prevent trying to register when `props.selectable_fd == 1` ?